ui/src/routes/results.tsx

```diff
@@ -14,30 +13,16 @@ const PAGE_SIZE = 10;
 
 function Results() {
   const searchTerm = useAppSelector((state) => state.search.searchTerm);
-  const [lectures, setLectures] = useState([]);
+  
   const [total, setTotal] = useState<number>(0);
   const [currentPage, setCurrentPage] = useState<number>(1);
   const [pageSize, setPageSize] = useState<number>(PAGE_SIZE);
-  const [loading, setLoading] = useState(false);
-
-  const searchLectures = async () => {
-    try {
-      setLoading(true);
-      const searchQuery = searchTerm
-        ? `?search=${searchTerm}&page=${currentPage}&page_size=${pageSize}`
-        : `?page=${currentPage}&page_size=${pageSize}`;
-      const results = await getApiRoot().get(`/search/lectures/${searchQuery}`);
-      setLoading(false);
-      setLectures(results.data.results);
-      setTotal(results.data.count);
-    } catch (error) {
-      setLectures([]);
-    }
-  };
+  
+  const searchQuery = searchTerm
+    ? `?search=${searchTerm}&page=${currentPage}&page_size=${pageSize}`
+    : `?page=${currentPage}&page_size=${pageSize}`;
 
-  useEffect(() => {
-    searchLectures();
-  }, [searchTerm, currentPage, pageSize]);
+  const { data, error, isLoading } = useGetLecturesQuery(searchQuery);
```

Comment:
When designing an API client you should hide implementation details behind it -> instead of giving the searchQuery to the function, you should give all the necessary data and then build the query in the `useGetLecturesQuery` function

Reply:
I followed the docs approach of reusable query https://redux-toolkit.js.org/tutorials/rtk-query#use-the-query-in-a-component, where you pass the desired parameter directly to the function.